llvm/tools/llvm-mca/llvm-mca.cpp

```diff
@@ -225,10 +225,51 @@ static cl::opt<unsigned> StoreQueueSize("squeue",
                                         cl::desc("Size of the store queue"),
                                         cl::cat(ToolOptions), cl::init(0));
 
-static cl::opt<bool>
-    PrintInstructionTables("instruction-tables",
-                           cl::desc("Print instruction tables"),
-                           cl::cat(ToolOptions), cl::init(false));
+enum class InstructionTablesType { NONE, NORMAL, FULL };
+
+class InstructionTablesOptionParser
+    : public cl::parser<enum InstructionTablesType> {
+public:
+  explicit InstructionTablesOptionParser(cl::Option &O)
+      : cl::parser<enum InstructionTablesType>(O) {}
+
+  bool parse(cl::Option &O, StringRef ArgName, StringRef Arg,
+             enum InstructionTablesType &Value) {
+    if (Arg.empty()) {
+      Value = InstructionTablesType::NORMAL;
+      return false;
+    }
+    return cl::parser<enum InstructionTablesType>::parse(O, ArgName, Arg,
+                                                         Value);
+  }
+};
```

Comment:
instead of using custom parser, I would recommend adding a `cl::EnumValN(InstructionTablesType::NORMAL, "", "")` to the list of `cl::values` in the declaration of `InstructionTablesOption`. So that it'll choose `NORMAL` by default in the absent of value.